pandas/io/excel/_base.py

```diff
@@ -598,41 +601,49 @@ def parse(
                                 data[row][col] = last
                             else:
                                 last = data[row][col]
-
+            future_warnings = []
             # GH 12292 : error when read one empty column from excel file
             try:
-                parser = TextParser(
-                    data,
-                    names=names,
-                    header=header,
-                    index_col=index_col,
-                    has_index_names=has_index_names,
-                    squeeze=squeeze,
-                    dtype=dtype,
-                    true_values=true_values,
-                    false_values=false_values,
-                    skiprows=skiprows,
-                    nrows=nrows,
-                    na_values=na_values,
-                    skip_blank_lines=False,  # GH 39808
-                    parse_dates=parse_dates,
-                    date_parser=date_parser,
-                    thousands=thousands,
-                    comment=comment,
-                    skipfooter=skipfooter,
-                    usecols=usecols,
-                    mangle_dupe_cols=mangle_dupe_cols,
-                    **kwds,
-                )
+                # Gotta catch deprecation warnings to raise at correct stacklevel :(
+                with warnings.catch_warnings(record=True) as w:
```

Comment:
We have find_stack_level for this

Reply:
This won't work unfortunately, since the ``read_excel`` and ``read_csv`` have different stacklevels but go through the same code. Currently, I'm catching the warning and raising it again with the correct stacklevel.

Reply:
Why does this not work? The find_stack_level function is designed to handle these cases

Reply:
agree, its possible that `find_stack_level` needs modification itself to handle this

Reply:
Tried it for read_excel, this seems to work as is.

Reply:
Whoops, I see what you mean now. It needs a patch to work for read_csv, though.